src/main/java/org/opensearch/neuralsearch/processor/TextImageEmbeddingProcessor.java

```diff
@@ -0,0 +1,218 @@
+            texts.put(INPUT_IMAGE, knnKeyMap.get(fieldMap.get(IMAGE_FIELD_NAME)));
+        }
+        return texts;
+    }
+
+    @VisibleForTesting
+    Map<String, String> buildMapWithKnnKeyAndOriginalValue(IngestDocument ingestDocument) {
+        Map<String, Object> sourceAndMetadataMap = ingestDocument.getSourceAndMetadata();
+        Map<String, String> mapWithKnnKeys = new LinkedHashMap<>();
+        for (Map.Entry<String, String> fieldMapEntry : fieldMap.entrySet()) {
+            String originalKey = fieldMapEntry.getValue(); // field from ingest document that we need to sent as model input, part of
+                                                           // processor definition
+
+            if (!sourceAndMetadataMap.containsKey(originalKey)) {
+                continue;
+            }
+            if (!(sourceAndMetadataMap.get(originalKey) instanceof String)) {
+                throw new IllegalArgumentException("Unsupported format of the field in the document, value must be a string");
+            }
+            mapWithKnnKeys.put(originalKey, (String) sourceAndMetadataMap.get(originalKey));
+        }
+        return mapWithKnnKeys;
+    }
+
+    @SuppressWarnings({ "unchecked" })
+    @VisibleForTesting
+    Map<String, Object> buildTextEmbeddingResult(String knnKey, List<Float> modelTensorList) {
+        Map<String, Object> result = new LinkedHashMap<>();
+        result.put(knnKey, modelTensorList);
+        return result;
+    }
+
+    private void validateEmbeddingFieldsValue(IngestDocument ingestDocument) {
+        Map<String, Object> sourceAndMetadataMap = ingestDocument.getSourceAndMetadata();
+        for (Map.Entry<String, String> embeddingFieldsEntry : fieldMap.entrySet()) {
+            Object sourceValue = sourceAndMetadataMap.get(embeddingFieldsEntry.getKey());
+            if (sourceValue != null) {
+                String sourceKey = embeddingFieldsEntry.getKey();
+                Class<?> sourceValueClass = sourceValue.getClass();
+                if (List.class.isAssignableFrom(sourceValueClass) || Map.class.isAssignableFrom(sourceValueClass)) {
+                    validateNestedTypeValue(sourceKey, sourceValue, () -> 1);
+                } else if (!String.class.isAssignableFrom(sourceValueClass)) {
+                    throw new IllegalArgumentException("field [" + sourceKey + "] is neither string nor nested type, can not process it");
+                } else if (StringUtils.isBlank(sourceValue.toString())) {
+                    throw new IllegalArgumentException("field [" + sourceKey + "] has empty string value, can not process it");
+                }
+            }
+        }
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private void validateNestedTypeValue(String sourceKey, Object sourceValue, Supplier<Integer> maxDepthSupplier) {
+        int maxDepth = maxDepthSupplier.get();
+        if (maxDepth > MapperService.INDEX_MAPPING_DEPTH_LIMIT_SETTING.get(environment.settings())) {
```

Comment:
Is the setting dynamic? Reading setting from environment might not get updated dynamically. Need to check.

Reply:
yes, that's dynamic

Reply:
Could you verify that dynamic update is actually getting applied to this processor? I think you should read setting from `ClusterService`?

Reply:
Good catch, with current code we are not getting dynamic updates for settings. Just pushed a commit to address that, we're using ClusterService to read than in runtime for each query